src/Hadolint/Formatter/Checkstyle.hs

```diff
@@ -78,8 +78,15 @@ escape = Text.concatMap doEscape
     doEscape c =
       if isOk c
         then Text.singleton c
-        else "&#" <> Text.pack (show (ord c)) <> ";"
-    isOk x = any (\check -> check x) [isAsciiUpper, isAsciiLower, isDigit, (`elem` [' ', '.', '/'])]
+        else xmlEscape c
```

Comment:
Maybe you'd like to use [one of the libraries instead](https://mojoauth.com/escaping/xml-escaping-in-haskell/)?

Reply:
That's probably overall the best way to go.
I had thought about it but wasn't quite sure which library was best to choose.
I've now decided to go with xml-conduit. It was a quite a bit of work to move the code over, but I'm happy as there is now overall less code and the bits that are there seem less fragile to me.